### ISSUE TITLE
Add a "primary source" variable

### ISSUE THREAD

**Issue by badumont** [citation-style-language/schema#388]
In the field of history, there is no uniform rules for citing primary sources in bibliography. The problem is that such items have two main titles: that of the source itself, and that of the publication or the manuscript through which we have access to it. Several strategies are used:
  * Cite only primary source's author and title, as is common with classics: if I understand well, this is what the new `classic` type will be made for. However it concerns only a handful of well-referenced Greek and Latin authors;
  * Cite primary source's author and title, then the edition's reference *without the title*. That's what is privileged by the [ACHCByz](http://achcbyz.com/achcbyzV2/normesEditoriales.html) (for instance, the title of Rampolla's edition of *Vita S. Melaniae Iunioris* is *Santa Melania giuniore, senatrice romana : Documenti contemporanei e note*, and it is stripped);
  * Cite primary source's author only, then the complete edition's reference. See for instance [Byzantine and Modern Greek Studies](https://www.cambridge.org/core/journals/byzantine-and-modern-greek-studies/information/instructions-contributors) (the edition's title is before the editor's name) and the [Jarhbuch der Österreichischen Byzantinistik](https://www.byzneo.univie.ac.at/fileadmin/user_upload/i_byzneo/JOEB/JOEB_Zitierregeln_deutsch_2015.pdf);
  * Cite both. It is stated explicitly in the guidelines of the [Revue des Études byzantines](https://www.icp.fr/recherche/chaires-et-instituts/institut-francais-detudes-byzantines-ifeb/revue-des-etudes-byzantines) (click on "normes éditoriales", as the direct link will soon change), but it is more widely practiced, albeit not always documented, for instance by the *Jarhbuch der Österreichischen Byzantinistik* (see [here](https://www.austriaca.at/0xc1aa5576%200x003ae9bf.pdf) n. 9 and 113).

The case of the classics apart, all these primary sources should be registered with the type of the work in which they are published. However, there is no way to represent them in CSL consistently across publishers: in some cases, the value of the `title` variable should be the source's title, in others the publication's title, while CSL currently doesn't provide any satisfactory solution to register both in a same bibliographical item.

This problem could be solved by creating a `primary-source` field. Adopting it would imply some rewriting for the styles that already exist, but it would also remove the need to use ad hoc workarounds: for instance, the [BMGS style](https://www.zotero.org/styles/byzantine-and-modern-greek-studies?source=1) requires to set the variable `original-publisher-place` in primary sources items in order to display properly the name of primary sources' authors. It would also provide a way to implement proper formatting of the primary sources in styles that barely ignore the issue (like the [JÖB style](https://www.zotero.org/styles/jahrbuch-der-osterreichischen-byzantinischen-gesellschaft?source=1) or the [REB style](https://www.zotero.org/styles/revue-des-etudes-byzantines?source=1), whose author gave up before having managed to meet the journal's requirements).

I hope that I clearly explained the issue. Please don't hesitate to require some clarifications if I didn't.

**Comment by bwiernik** [comment]
I am really not following what the styling you are saying is needed about titles. Can you give some specific real examples?

**Comment by badumont** [comment]
Yes, I should have done it before. Say that we want to refer to Anastasius Sinaita's *Hodegos*, edited by Karl-Heinz Uthemann in the volume 8 of the CCSG collection under the title *Anastasii Sinaitae Viae Dux* (*viae dux* is the translation of *hodegos* in Latin). Here are the rendering possibilities :
  * ACHCByz: ANASTASTASIUS SINAITA, *Hodegos*, ed. K.-H. UTHEMANN (CCSG 8), Tournhout 1980
  * BMGS: Anastasius Sinaita, *Anastasii Sinaitae Viae Dux*, ed. K.-H. UTHEMANN [CCSG 8] (Tournhout 1980)
  * REB: ANASTASIUS SINAITAI, *Hodegos*, in K.-H. UTHEMANN (ed.), *Anastasii Sinaitae Viae Dux* (CCSG 8), Tournhout 1980

So if one plans to write for the BMGS, one should register *Anastasii Sinaitae Viae Dux* as the `title` variable and let *Hodegos* alone, and conversely if one writes for a publication of the ACHCByz — not to mention the REB, which requires both. One can not use the `container-title` variable to this end, as primary sources can also be published in journal articles and in chapters in collective publications.

**Comment by badumont** [comment]
Yes, I should have done it before. Say that we want to refer to Anastasius Sinaita's *Hodegos*, edited by Karl-Heinz Uthemann in the volume 8 of the CCSG collection under the title *Anastasii Sinaitae Viae Dux* (*viae dux* is the translation of *hodegos* in Latin). Here are the rendering possibilities :
  * ACHCByz: ANASTASTASIUS SINAITA, *Hodegos*, ed. K.-H. UTHEMANN (CCSG 8), Tournhout 1980
  * BMGS: Anastasius Sinaita, *Anastasii Sinaitae Viae Dux*, ed. K.-H. UTHEMANN [CCSG 8] (Tournhout 1980)
  * REB: ANASTASIUS SINAITAI, *Hodegos*, in K.-H. UTHEMANN (ed.), *Anastasii Sinaitae Viae Dux* (CCSG 8), Tournhout 1980

So if one plans to write for the BMGS, one should register *Anastasii Sinaitae Viae Dux* as the `title` variable and let *Hodegos* alone, and conversely if one writes for a publication of the ACHCByz — not to mention the REB, which requires both. One can not use the `container-title` variable to this end, as primary sources can also be published in journal articles and in chapters in collective publications.

**Comment by bwiernik** [comment]
Here are some of my thoughts:

The item `title` should be the title that would be used to retrieve the item. For those examples, it would be the actual title of the work. 

Likewise, `container-title` should hold the book title or journal title. That is a consistent usage with other item types. 

For the case where you have a primary source published as a component of an article or chapter, the article/chapter title should be used as the `title` and `part-title` (new in CSL 1.0.2) should be used to hold the title of the primary source. 

The above creates maximal compatibility across styles and disciplines. As for item types, I’m inclined to think that these should be entered as either `article-journal` or `chapter` depending on how they are published. `manuscript` or `pamphlet` are also options (depending on the type of source, also `graphic` or `song`), though that might make some mess if the items are published in a journal vs a book (the latter format generally being expected). For items published within a larger article or chapter, `figure` seems appropriate (but see the issue below). 

(We haven’t much discussed the issue of citing a component of an article or chapter. This could potentially be relevant for, eg, the item types `song`, `graphic`, or `figure`. This hasn’t really come up before because most fields would just cite the article, for example, rather than the figure, and would refer to the figure the same way as a page number. @denismaier it seems odd that a `figure` item would have its `title` being the title of the article containing it. It seems like a variable like `article-title`/`chapter-title` [only one] for the title of the article/chapter/entry containing the item would be good for `figure` and similar item types. We already have a `chapter-number` variable, so `chapter-title` seems like a natural complement, but I’m wonder if `article-title` might engender less confusion?)

tl;dr

**Comment by denismaier** [comment]
Argh, this reminds me of our discussions we've had regarding the treatment of introductions in Chicago styles: @cormacrelf proposed [here](https://discourse.citationstyles.org/t/introductions-and-other-supplementary-material/1573/4?u=denis_maier) to add some sort of "first-class, namespaced escape hatches" in order to cover this. And I think we're in a very similar situation here. We can cover 98% of these things, but style interoperability may suffer in certain cases.

Regarding these examples:

    ACHCByz: ANASTASTASIUS SINAITA, Hodegos, ed. K.-H. UTHEMANN (CCSG 8), Tournhout 1980
    BMGS: Anastasius Sinaita, Anastasii Sinaitae Viae Dux, ed. K.-H. UTHEMANN [CCSG 8] (Tournhout 1980)
    REB: ANASTASIUS SINAITAI, Hodegos, in K.-H. UTHEMANN (ed.), Anastasii Sinaitae Viae Dux (CCSG 8), Tournhout 1980

I think ACHCByz and REB can be covered this way:

```yaml
title: Hodegos
container-title: Anastasii Sinaitae Viae Dux
```

But if you change to BMGS, how will you use the `container-title` instead of the title in this case, without doing it in all cases, which you will most likely not want? Maybe we can somehow use the `part-title` here, but I'm not sure.

The bigger problem with all this is that these kind of rules make distinctions not based on item metadata, but on the content of an item.

(I agree adding a new `article-title` or `chapter-title` or some other `intermediary-container` would be a good move.)

**Comment by badumont** [comment]
Thank you both for these very interesting insights. I totally agree that a primary source should be of the type of the work in which it has been published. 

If I understand well, whereas I tended to consider the primary source title as a supplementary information, you prefer to set it as the item's main title, hence the need to add `article-title` and `chapter-title` variables that would anyway be required by the `figure` type. I think that it is a better solution than my initial proposal, since it would lead to a better display in reference managers. 

I would prefer the adding of new `*-title` variables for sources published in chapters or articles than registering the publication title in `title` and the source in `title-part`, since this would reduce the homogeneity of encoding strategies across item types for primary sources — and hence comprehensibility for end users. 

I don't think that there would be a problem for BMGS if we use `title` and `container-title` as you propose to do. We could imagine the following code for REB (restricting to book and chapter types and setting aside information after the title):

```xml
   <macro name="author-source">
    <choose>
      <if variable="chapter-title">
        <names variable="author"/>
      </if>
      <else-if type="book" variable="container-title" match="all">
        <names variable="author"/>
      </else-if>
    </choose>
  </macro>

  <macro name="title-source">
    <choose>
      <if variable="chapter-title">
        <text variable="title" font-style="italic"/>
      </if>
      <else-if type="book" variable="container-title" match="all">
        <text variable="title" font-style="italic"/>
      </else-if>
    </choose>
  </macro>

  <macro name="author">
    <names variable="author"/>
    <names variable="editor translator">
      <label form="short" suffix=" "/>
      <name/>
    </names>
  </macro>

  <macro name="title">
    <choose>
      <if variable="chapter-title">
        <text variable="chapter-title"/>
      </if>
      <else-if type="book" variable="container-title" match="all">
        <text variable="container-title" font-style="italic"/>
      </else-if>
      <else>
        <text variable="title"/>
      </else>
    </choose>
  </macro>

  <citation>
    <layout delimiter=", " suffix=".">
      <text macro="author-source"/>
      <text macro="title-source"/>
      <text macro="author"/>
      <text macro="title"/>
    </layout>
  </citation>
```

And for BMGS:

```xml
  <macro name="author-source">
    <choose>
      <if variable="chapter-title">
        <names variable="author"/>
      </if>
      <else-if type="book" variable="container-title" match="all">
        <names variable="author"/>
      </else-if>
    </choose>
  </macro>

  <macro name="title-source">
    <choose>
      <if variable="chapter-title">
        <text variable="chapter-title" font-style="italic"/>
      </if>
      <else-if type="book" variable="container-title" match="all">
        <text variable="container-title" font-style="italic"/>
      </else-if>
    </choose>
  </macro>

  <macro name="author">
    <names variable="author"/>
    <names variable="editor translator">
      <label form="short" suffix=" "/>
      <name/>
    </names>
  </macro>

  <macro name="title">
    <choose>
      <if variable="chapter-title"/>
      <else-if type="book" variable="container-title" match="all"/>
      <else>
        <text variable="title"/>
      </else>
    </choose>
  </macro>

  <citation>
    <layout delimiter=", " suffix=".">
      <text macro="author-source"/>
      <text macro="title-source"/>
      <text macro="author"/>
      <text macro="title"/>
    </layout>
  </citation>
```

In other terms, the presence of a `container-title` variable for an item of the `book` type is sufficient to indicate that the item is a primary source and should be handled accordingly (since I don't think that it would be set in a `book` item, nor `chapter-title` in a `chapter` item, for other purposes).

The only advantage I still find in creating a `primary-source` variable is that it would not impose to modify existing CSL that don't need to output the primary source's original title and thus already manage to handle correctly the citation of primary sources, like BMGS. It would also require less modifications to existing users' databases (only adding the `primary-source` variable for those who need it vs changing the `title` variable and adding `*-title` variables for all).

**Comment by denismaier** [comment]
> In other terms, the presence of a `container-title` variable for an item of the `book` type is sufficient to indicate that the item is a primary source and should be handled accordingly (since I don't think that it would be set in a `book` item, nor `chapter-title` in a `chapter` item, for other purposes).

Well, we have actually decided to use this for items that originally have been published as independent items, but are are now contained in a container if some sort, a book or a multivolume book or so. A typical case is a book now republished in the Collected writings of an author, think of Adorno's Negative dialectics, Marx's Capital. The title of those works should be in italics, not in quotes. That's a similar case, but not strictly identical.

**Comment by badumont** [comment]
Oh, I see. Could it lead to a different formatting? In which case it would be an additional argument for the creation of a special variable for primary sources titles. 

Another argument that I haven't cited yet would be the greater simplicity and readability of conditionals in CSL stylesheets, as only a `<if variable="primary-source">`  element would be required to handle the case of primary sources.

**Comment by bwiernik** [comment]
How would BMGS cite a typical chapter, rather than a primary source chapter? Does it also omit the chapter title there?

**Comment by denismaier** [comment]
Here is the example given in the style sheet:
> P. Golden, ‘The peoples of the Russian forest belt’, in D. Sinor (ed.), The Cambridge History of Early Inner Asia(Cambridge 1990)256–8.

**Comment by denismaier** [comment]
What would speak for the addition of a `primary-source` (or similar) variable is that these titles may not appear on the titlepage of a book or in bibliographic metadata. (However, the same may be true for the author's name.) And you will most likely not find those works in library catalogs when you search for this commonly used name. E.g. your example looks like this in our catalog: https://swisscovery.slsp.ch/permalink/41SLSP_NETWORK/1ufb5t2/alma991073936179705501 Here, hodegos appears somewhere in the annotations, but not in the title that is just "Anastasii Sinaitae Viae dux". So one question if we should give priority to the bibliographic title or the title by convention.

Anyway, in CSL 1.1 we'll extend the capabilities of title variables by converting them to objects. Maybe this use case could be covered there?

**Comment by bwiernik** [comment]
I am concerned that `primary-source` is very niche to history as a field, and this isn't how such items would be treated or cited in other disciplines. Let's investigate a bit more things like library catalogs that @denismaier is looking into. That should be helpful. Based on that example, `part-title` seems like it would cover use cases well.

**Comment by denismaier** [comment]
> I am concerned that `primary-source` is very niche to history as a field, and this isn't how such items would be treated or cited in other disciplines.

Exactly. My thinking above was that we could add it as an alternative title once title variables are objects. Just as we do with screen names, alternative names, or the language alternates proposed here: https://github.com/citation-style-language/schema/issues/327#issuecomment-667003188

**Comment by badumont** [comment]
I understand your concern. Setting it as an alternate inside a title object seems to be a nice solution, presenting the advantages discussed above without being too specific. Actually it would conform to libraries' catalogues, which either ignore those titles or register it as "alternative" or "other title". What type of alternate would you consider? Something like `description`, `content` or `meta`?

As for using the `part-title` variable, I fear that it conflict with other needs (like the one for which `part-title` and `part-number` are originally intended...). I know primary sources that were published in multi-part articles. The problem with using an existing variable in a somehow non-standard way is that we risk to need it for the work it is published in and whose references we have to give in full after the primary source title. That's why, for my custom style, I ended up putting the source title in the `abstract` field, being sure that I would never use it otherwise... One could satisfy oneself with it until CSL 1.1 adds an alternate mechanism.

**Comment by bwiernik** [comment]
As a stopgap field, I'd suggest a less-used field like `annote` versus `abstract`, which comes pre-filled for many items when you import from the web.

**Comment by badumont** [comment]
Thank you for the suggestion! Unfortunately it doesn't appear in Zotero, but that is not an obstacle per se.

**Comment by denismaier** [comment]
If it's for your own purposes only you can even use arbitrary variables via extra. Currently that should work with pandoc and citeproc-js. Don't know about citeproc-rs that will replace citeproc-js at some point.
________________________________________
Von: badumont <notifications@github.com>
Gesendet: Dienstag, 22. Dezember 2020 19:13:08
An: citation-style-language/schema
Cc: Maier, Denis Christian (UB); Mention
Betreff: Re: [citation-style-language/schema] Add a "primary source" variable (#388)

Thank you for the suggestion! Unfortunately it doesn't appear in Zotero, but that is not an obstacle per se.

**Comment by bwiernik** [comment]
@badumont In Zoter, add it to the top of Extra like this:
`annote: Primary source title`

@denismaier By the point that Zotero moves to citeproc-rs, it will have implemented its own migration from Extra/custom field system.

**Comment by badumont** [comment]
Actually I am refactoring all my code and writing a user guide because I know too many colleagues who can't use Zotero because of their lack of knowledge of its advanced features and the inavailability of styles filling their needs. My goal would be to create a set of styles that could ultimately go to the official repository, so I don't want to add custom extensions to the CSL guidelines, although I don't know of such baroque a use of its variables would permit to do it for now...
